.cascade-code/Chapman.edu/test-section/tatwell/query_api_spike/query_api.vtl

```diff
@@ -0,0 +1,175 @@
+#*
+ * query_api.vtl
+ * Test format to demonstrate usage of Cascade locator tool and Query API.
+ *
+ * For more info, see spike user story:
+ * https://trello.com/c/InwKZ4w9
+ *#
+
+#import( "/_cascade/formats/helpers/debug.velocity" )
+
+##
+## Hardcoded Params
+##
+
+
+##
+## Vars
+##
+## Cascade provides a $currentPage object to every format.
+
+## This can be selected when Page Block is present:
+##set ( $pageNode = $_XPathTool.selectSingleNode($contentRoot, "//system-page[@current = 'true']") )
```

Comment:
This is what we currently use in most cases to set `$currentPage`. Two drawbacks to this:

- It requires a block such as Page Block.
- It overwrites the `$currentPage` Cascade object that Cascade provides for free in every format.

Reply:
I'm not sure that's quite true. For purposes of the LeftNav_velocity format we use the big LeftNav index block and from that we get the current page thru the variable you note above. We don't do a second call to another block such as Current Page in order to get that. A format can only act on one block at a time if using XPath